consensus/safety-rules/src/safety_rules.rs

```diff
@@ -462,6 +466,34 @@ impl SafetyRules {
         let signature = self.sign(timeout)?;
         Ok(signature)
     }
+
+    fn guarded_sign_commit_vote(
+        &mut self,
+        ledger_info: LedgerInfoWithSignatures,
+        verifier: &ValidatorVerifier,
```

Comment:
this is not what's expected, the coming ledger_info contains a BlockInfo with empty state id, but the outgoing signature needs to sign on the real executed state

Reply:
Resolved.

Reply:
we don't pass the verifier here, you can get it from self.epoch_state()

Reply:
Thanks